src/components/Geogebra/GeogebraAdminView.js

```diff
@@ -0,0 +1,255 @@
+import feathersClient from '../../feathers-client';
+
+const api = feathersClient;
+
+const Consts = {
+    // Ownership indicators
+
+    UNASSIGNED: 'unassigned',
+    ADMIN: 'admin', // ~teacher
+    USERNAME: 'username',
+
+    // Geogebra specific options
+
+    // Caption style related to label style
+    // determining how objects' names are displayed in
+    // the applet. Caption style means that objects are
+    // named by their captions.
+    CAPTION_STYLE: 3,
+
+    POINT: 'point',
+    TEXTFIELD: 'textfield',
+    NUMERIC: 'numeric',
+
+    /**
+     * GGB commands have the following
+     * format: <object-label>:<command-body>.
+     *
+     * @param {String} label
+     * @param {String} cmdStrBody
+     */
+    getCommand(label, cmdStrBody) {
+        return `${label}:${cmdStrBody}`;
+    },
+};
+
+class GeogebraAdminView {
```

Comment:
You are duplicating a lot of code between `GeogebraAdminView` and `GeogebraMergedAdminView`. For example, why don't you extract `Consts`? It very safe to do so. More importantly, however - include only what you need. You dont need to check locks because all elements will be always locked - just set the lock when added. _DRY_.

Reply:
You suggested bottom-to-top approach and gave me a notice that in JS there should be no class inheritance :)